### DISCUSSION TITLE
Required Link Text only if there is a linkValue

### DISCUSSION THREAD

**Discussion by samuel-clara** [verbb/hyper#192] [General]
Is there a way to make the linkText field required only if the linkValue is not empty?
A good alternative could be an empty link type, but there is currently no way to have a new link type without linkValue.

**Comment by engram-design** [comment]
Great point that there's not really a way at the moment. It's indeed arguable that a link block shouldn't be considered "live" unless it has a link value, therefore doesn't need to be validated. I'll play around with options.

Related to https://github.com/verbb/hyper/issues/8 for better empty-state handling.

**Comment by engram-design** [comment]
Updated in [3.0.0-beta.1](https://github.com/verbb/hyper/releases/tag/3.0.0-beta.1).
